test/ui/CategoryFacet/CategoryFacetSearchTest.ts

```diff
@@ -0,0 +1,191 @@
+import { CategoryFacet } from '../../../src/ui/CategoryFacet/CategoryFacet';
+import { mock, basicComponentSetup } from '../../MockEnvironment';
+import { FakeResults } from '../../Fake';
+import { CategoryFacetSearch } from '../../../src/ui/CategoryFacet/CategoryFacetSearch';
+import { CategoryFacetQueryController } from '../../../src/controllers/CategoryFacetQueryController';
+import _ = require('underscore');
+import { IGroupByValue } from '../../../src/rest/GroupByValue';
+import { $$ } from '../../../src/Core';
+import { KEYBOARD } from '../../../src/utils/KeyboardUtils';
+
+export function CategoryFacetSearchTest() {
+  describe('CategoryFacetSearch', () => {
+    let categoryFacetMock: CategoryFacet;
+    let categoryFacetSearch: CategoryFacetSearch;
+    let realDebounce;
+    let fakeGroupByValues: IGroupByValue[];
+    beforeEach(() => {
+      realDebounce = _.debounce;
+      spyOn(_, 'debounce').and.callFake(function(func) {
```

Comment:
What's that for ?

Reply:
`jasmine.clock().tick()` doesn't play well with undescore's debounce. I used this to make the debounce call trigger asap